x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalVerifier.java

```diff
@@ -13,26 +13,29 @@
 import org.elasticsearch.xpack.esql.plan.logical.Enrich;
 import org.elasticsearch.xpack.esql.plan.logical.LogicalPlan;
 
+import static org.elasticsearch.xpack.esql.common.Failure.fail;
+import static org.elasticsearch.xpack.esql.core.expression.Attribute.datatypeEquals;
+
 public final class LogicalVerifier {
 
     public static final LogicalVerifier INSTANCE = new LogicalVerifier();
 
     private LogicalVerifier() {}
 
     /** Verifies the optimized logical plan. */
-    public Failures verify(LogicalPlan plan, boolean skipRemoteEnrichVerification) {
+    public Failures verify(LogicalPlan planAfter, boolean skipRemoteEnrichVerification, LogicalPlan planBefore) {
```

Comment:
nit/bikeshed: naming could be a little more explicit, maybe:
```suggestion
    public Failures verify(LogicalPlan optimizedPlan, boolean skipRemoteEnrichVerification, LogicalPlan planBeforeOptimization) {
```

Also, since we only need the output of the non-optimized plan, we could consider slimming down the signature to
```
    public Failures verify(LogicalPlan optimizedPlan, boolean skipRemoteEnrichVerification, List<Attribute> expectedOutputAttributes)
```

Reply:
Done and done